ipa-core/src/net/http_serde.rs

```diff
@@ -294,40 +272,16 @@ pub mod query {
             }
         }
 
-        #[async_trait]
-        impl FromRequest<hyper::Body> for Request {
-            type Rejection = Error;
-
-            async fn from_request(
-                req: &mut RequestParts<hyper::Body>,
-            ) -> Result<Self, Self::Rejection> {
-                let Path(query_id) = req.extract().await?;
-                let QueryConfigQueryParams(config) = req.extract().await?;
-                let Json(RequestBody { roles }) = req.extract().await?;
-                Ok(Request {
-                    data: PrepareQuery {
-                        query_id,
-                        config,
-                        roles,
-                    },
-                })
-            }
-        }
-
         #[derive(Serialize, Deserialize)]
-        struct RequestBody {
-            roles: RoleAssignment,
+        pub struct RequestBody {
```

Comment:
curious what is causing it to be `pub` now? is it the new Axum requirement?

Reply:
So, in the past we used to have a custom request extractor for each API, which would in turn refer these (internal to it) Request objects. That extractor used to be `pub`. Since we got rid of those now the handlers need to see these `Requests`.